parser/src/main/scala/hmda/parser/fi/lar/LarCsvParser.scala

```diff
@@ -83,63 +87,67 @@ object LarCsvParser {
         )
       val denial = Denial(denial1, denial2, denial3)
 
-      Right(LoanApplicationRegister(
-        id,
-        respId,
-        agencyCode,
-        loan,
-        preapprovals,
-        actionType,
-        actionDate,
-        geography,
-        applicant,
-        purchaserType,
-        denial,
-        rateSpread,
-        hoepaStatus,
-        lienStatus
-      ))
+      Right(
+        LoanApplicationRegister(
+          id,
+          respId,
+          agencyCode,
+          loan,
+          preapprovals,
+          actionType,
+          actionDate,
+          geography,
+          applicant,
+          purchaserType,
+          denial,
+          rateSpread,
+          hoepaStatus,
+          lienStatus
+        )
+      )
     } else {
-      Left(larErrors)
+      val lErrors = validation.toEither.left.get
+      Left(lErrors.list.toList)
```

Comment:
I'm wondering about this too. Is it possible to use a `match` statement on `larErrors`, instead of an `if`/`else` statement on the disjunction? I bet that would make it easier to extract the errors from that validation, and it might be more "idiomatic".